src/main/java/com/redhat/devtools/lsp4ij/LSPFileListener.java

```diff
@@ -68,17 +68,27 @@ public void fileClosed(@NotNull FileEditorManager source, @NotNull VirtualFile f
 
     @Override
     public void propertyChanged(@NotNull VirtualFilePropertyEvent event) {
-        if (event.getPropertyName().equals(VirtualFile.PROP_NAME) && event.getOldValue() instanceof String) {
-            // A file (Test1.java) has been renamed (to Test2.java) by using Refactor / Rename from IJ
+        // Either a file (Test1.java) has been renamed (to Test2.java) by using Refactor / Rename from IJ
+        // or a "properties" file is changed or saved.
+        if (isRenameFile(event)) {

```

Comment:
`if (!isRenameFile(event)) {`

Reply:
Once you will fix that, I think the PR can be merged.

It would be really nice to write tests with this usecase but I need to find time to write them.

Reply:
Agreed, I reformatted the code also.

Reply:
I tested this with our tools to make sure this change is correct.